python/llm/src/bigdl/llm/transformers/models/baichuan2.py

```diff
@@ -70,6 +70,23 @@ def baichuan_13b_rms_norm_forward(self, hidden_states):
         return self.weight * hidden_states.to(input_dtype)
 
 
+def baichuan_mlp_forward(
+    self,
+    x: torch.Tensor,
+) -> torch.Tensor:
+    if x.shape[1] == 1 and x.dtype == torch.float32 and x.device.type == 'xpu' \
+            and not (self.training and x.requires_grad):
+        import linear_q4_0
+        x_2d = x.view(-1, x.shape[-1])
+        if not x_2d.is_contiguous():
+            x_2d = x_2d.contiguous()
+        return self.down_proj(linear_q4_0.mlp_forward_q4_0_xpu(
```

Comment:
it seems linear_q4_0.mlp_forward_q4_0_xpu only support a fixed activation function ?